tokenizers/javascript/index.js

```diff
@@ -0,0 +1,88 @@
+const esprima = require('esprima')
+const escodegen = require('escodegen')
+const fs = require('fs-extra-promise')
+const tokenizer = require('./tokenizer')
+
+const immutable = require('immutable')
+const walk = require('esprima-walk')
+
+const { base64FileName } = require('./util')
+
+const estools = require('estools')
+
+const TOKENIZER_SCOPE_FILE = 'file-scope'
+const TOKENIZER_SCOPE_FUNCTION = 'function-scope'
+
+const TOKENIZER_SCOPE = TOKENIZER_SCOPE_FILE
+
+// TODO: estools map / filter / traverse (instead of walk)
+// - filter subfunctions from fuction asts somehow
+// - test on SCC
+
+// TODO: get rid of the function block and indentation
+const regenerateFunctionCode = function(functionAst) {
+  codegenOptions = { // NOTE: doesn't help
+    format: {
+      parentheses: false
+    }
+  }
+
+  // NOTE: functionAst.body ommits the function signature (returns block only)
+  return escodegen.generate(functionAst.body, {})
+}
+
+const processFile = function(fileName, data) {
+  //let parentId = base64FileName(fileName) // TODO: incorporate repo name / hash
+  let parentId = fileName
+  let blockId = 1
+
+  if (TOKENIZER_SCOPE === TOKENIZER_SCOPE_FILE) {
+    return immutable.List.of(tokenizer(data, parentId, blockId))
```

Comment:
let's increment the blockId before we call the tokenizer. Or do you increment it somewhere else?